Documentation/linux-install-diagnosis.md

```diff
@@ -0,0 +1,144 @@
+
+### Package feed not installed due to bug in `packages-microsoft-prod.deb`
+The output from the commands above should include a line like this:
+
+```
+/etc/apt/sources.list.d/microsoft-prod.list:deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.10/prod cosmic main
+```
+
+First, try reinstalling the package to work around a bug in an old version that improperly handles upgrades:
+
+```
+sudo dpkg --purge packages-microsoft-prod && sudo dpkg -i packages-microsoft-prod.deb
+sudo apt-get update
+sudo apt-get install dotnet-sdk-2.2
+```
+
+If that doesn't work, there are commands listed on each download page that show how to directly install the feed rather than using `packages-microsoft-prod.deb`. These commands are specific to your Linux distro and version.
+
+
+## `E: Unable to correct problems, you have held broken packages.`
+
+After running `apt-get install dotnet-sdk-2.2`, you may see:
+
+```
+$ apt-get install dotnet-sdk-2.2
+Reading package lists... Done
+Building dependency tree       
+Reading state information... Done
+Some packages could not be installed. This may mean that you have
+requested an impossible situation or if you are using the unstable
+distribution that some required packages have not yet been created
+or been moved out of Incoming.
+The following information may help to resolve the situation:
+
+The following packages have unmet dependencies:
+ dotnet-sdk-2.2 : Depends: aspnetcore-runtime-2.2 (>= 2.2.5) but it is not going to be installed
+                  Depends: dotnet-runtime-2.2 (>= 2.2.5) but it is not going to be installed
+E: Unable to correct problems, you have held broken packages.
+```
+
+This means `aspnetcore-runtime-2.2` and `dotnet-runtime-2.2` failed to install, but unfortunately it doesn't say why. You *can* follow the chain by running `apt-get install` for each unmet dependency, but it's usually safe to assume this error is because `dotnet-runtime-deps-2.2` failed to install.
+
+The following commands give more info about the error and the current state of your machine. Please run them and include the output in your issue report:
+
+```
+# This shows the package repositories you have configured. (Source: https://askubuntu.com/a/741948)
+grep -r --include '*.list' '^deb ' /etc/apt/sources.list /etc/apt/sources.list.d/
+
+# These commands show processor and Linux distribution information.
+uname -a
+cat /etc/os-release
+
+# This command shows why dotnet-runtime-deps-2.2 can't install.
+apt-get install dotnet-runtime-deps-2.2
+
+# You can copy this entire block of text and paste it into your terminal for convenience.
+```
+
+Next are common issues and resolutions:
+
+### Mismatched package repository vs. Linux distro/version
+The list of package repositories should include a `microsoft-prod.list` line, like this one:
+
+```
+/etc/apt/sources.list.d/microsoft-prod.list:deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.10/prod cosmic main
```

Comment:
Should make it clear that this is output from the previous grep command and is not the actual contents of microsoft-prod.list.